guja-core/src/main/java/com/wadpam/guja/oauth2/web/OAuth2Filter.java

```diff
@@ -83,28 +84,30 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
       request.setAttribute(NAME_ACCESS_TOKEN, accessToken);
 
       final DConnection conn = verifyAccessToken(accessToken);
-      if (null != conn) {
-        LOGGER.info("Authenticated");
-        LOGGER.debug(" userId = {} roles {}", conn.getUserId(), conn.getRoles());
 
-        // User is authenticated
-        request.setAttribute(NAME_CONNECTION, conn);
-        request.setAttribute(NAME_USER_ID, conn.getUserId());
-        request.setAttribute(NAME_ROLES, conn.getRoles());
+      // access_token used here for app authentication must be issued by self:
```

Comment:
They're not! A federated token must only be used when registering to get hold of the application-specific token. It must not be used subsequently to access the resources.